### ISSUE TITLE
MP4 Remuxed files used in Vegas Pro 19 causes VP19 to glitch out

### ISSUE THREAD

**Issue by Setnour6** [obsproject/obs-studio#7233]
### Operating System Info

Windows 10

### OBS Studio Version

Other

### OBS Studio Version (Other)

27.2.3

### OBS Studio Log URL

N/A

### OBS Studio Crash Log URL

N/A

### Expected Behavior

Vegas Pro 19 works as normal as the remuxed MP4 file is added onto the Vegas Pro 19 project.

### Current Behavior

Vegas Pro 19 glitches out moments after adding the remuxed MP4, and it does that when the MP4 is loaded in the project. Closing and opening the project does not work.
See this reddit post for a similar issue: https://www.reddit.com/r/VegasPro/comments/tut70o/vegas_glitching_vegas_pro_19/?utm_source=BD&utm_medium=Search&utm_name=Bing&utm_content=PSR1
Please see attached the log file
[2022-08-26 11-09-16.txt](https://github.com/obsproject/obs-studio/files/9456771/2022-08-26.11-09-16.txt)

### Steps to Reproduce

1. Record your video in MKV format
2. Remux your MKV to MP4
3. Drop your remuxed MP4 in a Vegas Pro 19 project
4. After a while, Vegas Pro 19 glitches out. This happens much faster when you select, move, change, edit, and even save.

### Anything else we should know?

OBS Studio Log itself is posted in the "Current Behaviour" section, as I recently updated OBS to the latest version (27.2.4, from 27.2.3) before making this issue.

FYI This is my first issue on the OBS Repository, so only take it with a smidge of spilled milk

**Comment by Setnour6** [comment]
This problem could more or less be a Vegas Pro 19 but I have no way to contact them, so here is the best place I could post this issue

**Comment by energizerfellow** [comment]
Do you have the same problems if you save directly to a MP4 container? How about MOV?

We'd also like to hear if adding `movflags=+empty_moov+frag_keyframe` in the muxer settings on a MP4 container breaks anything in recent versions of Vegas Pro. This should be tested separately from MP4 without any custom muxer options.

**Comment by Setnour6** [comment]
> Do you have the same problems if you save directly to a MP4 container? How about MOV?
Vegas Pro 19 doesn't support MOV files.
> We'd also like to hear if adding `movflags=+empty_moov+frag_keyframe` in the muxer settings on a MP4 container in OBS Studio breaks any media handling in recent versions of Vegas Pro. 
I don't know how to add tags like that.
>This should be tested separately from MP4 without any custom muxer options.
I don't know how to test it if Vegas Pro 19 won't accept MOV files.

**Comment by energizerfellow** [comment]
> I don't understand what you mean by that

Test by recording directly to a MP4 container in OBS Studio and see if said MP4s are stable in Vegas Pro. This is under Settings -> Recording -> Recording Format in OBS.

If the regular MP4 recordings are good in Vegas Pro, then further tests by adding `movflags=+empty_moov+frag_keyframe` to the Custom Muxer Settings box under Settings -> Output -> Recording -> Custom Muxer Settings in OBS, then saying OK. See if these fragmented MP4 recordings work in Vegas Pro.

**Comment by Setnour6** [comment]
> > I don't understand what you mean by that
> 
> Test by recording directly to a MP4 container in OBS Studio and see if said MP4s are stable in Vegas Pro. This is under Settings -> Recording -> Recording Format in OBS.
> 
> If the regular MP4 recordings are good in Vegas Pro, then further tests by adding `movflags=+empty_moov+frag_keyframe` to the Custom Muxer Settings box under Settings -> Output -> Recording -> Custom Muxer Settings in OBS, then saying OK. See if these fragmented MP4 recordings work in Vegas Pro.

Thank you for letting me know about this. I will try that out and see if it works, and also perform further testing with the Custom Muxer Settings. My guess if that I have to put the Custom Muxer Setting thing when recording under the MKV format.
I'll leave this open in case you or anyone else have anything else to say, and also if I want to update you on how it's going for me with the video recording.

**Comment by RytoEX** [comment]
> My guess if that I have to put the Custom Muxer Setting thing when recording under the MKV format.

No. You should add those settings *after* changing the container to MP4. We are looking for tests of both plain MP4 and fragmented MP4.

**Comment by Setnour6** [comment]
> > My guess if that I have to put the Custom Muxer Setting thing when recording under the MKV format.
> 
> 
> 
> No. You should add those settings *after* changing the container to MP4. We are looking for tests of both plain MP4 and fragmented MP4.

Oh okay. My guess then is plain MP4 is without custom muxer settings and fragmented is with the settings (forgive me, I'm not strong with OBS terminology). I'll try the recordings with and without the muxer setting inputs and see how it goes.

**Comment by gxalpha** [comment]
> I'll try the recordings with and without the muxer setting inputs and see how it goes.

Have you been able to test this?
